src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -76,29 +73,19 @@ public void processBlobs() {
 
     private void processZipFiles(CloudBlobContainer container) {
         log.info("Processing blobs for container {}", container.getName());
+        List<String> zipFilenames = getFileNames(container);
 
-        // Randomise iteration order to minimise lease acquire contention
-        // For this purpose it's more efficient to have a collection that
-        // implements RandomAccess (e.g. ArrayList)
-        List<String> zipFilenames = new ArrayList<>();
-        container
-            .listBlobs()
-            .forEach(b -> {
-                String fileName = FilenameUtils.getName(b.getUri().toString());
-                if (Strings.isNullOrEmpty(fileName)) {
-                    log.error("Cannot extract filename from list blob item. URI: {}", b.getUri());
-                } else {
-                    zipFilenames.add(fileName);
-                }
-            });
-        Collections.shuffle(zipFilenames);
         for (String zipFilename : zipFilenames) {
             tryProcessZipFile(container, zipFilename);
         }
 
         log.info("Finished processing blobs for container {}", container.getName());
     }
 
+    List<String> getFileNames(CloudBlobContainer container) {
+        return getZipFileNamesFromContainer(container);
+    }
```

Comment:
why? it can be unit tested on it's own no need for nested method

Reply:
In BlogProcessorTaskTest it has to be mocked. Tried to use PowerMock and JMockit for mocking static call, both have some issues when being used with JUnit5

Reply:
but why mock? can just use same thing as in utility unit test:

```java
given(container.listBlobs()).willReturn(singletonList(blob));
given(blob.getUri()).willReturn(URI.create("file.zip"));
```

Reply:
Ok.